modules/platform/nuxeo-automation/nuxeo-automation-core/src/main/java/org/nuxeo/ecm/automation/core/AutomationComponent.java

```diff
@@ -135,21 +136,25 @@ public void registerContribution(Object contribution, String extensionPoint, Com
                     widgetDefinitionList.add(widgetDescriptor.getWidgetDefinition());
                 }
             }
+            Class<?> type;
             try {
-                service.putOperation(opc.type, opc.replace, contributor.getName().toString(), widgetDefinitionList);
+                type = Class.forName(opc.type);
+            } catch (ClassNotFoundException e) {
+                throw new IllegalArgumentException("Invalid operation class '" + opc.type + "': class not found.");
+            }
+            try {
+                service.putOperation(type, opc.replace, contributor.getName().toString(), widgetDefinitionList);
```

Comment:
What do you think about merging the two try/catch? Something like below:

```java
            try {
                Class<?> type = Class.forName(opc.type);
                service.putOperation(type, opc.replace, contributor.getName().toString(), widgetDefinitionList);
            } catch (ClassNotFoundException e) {
                throw new IllegalArgumentException("Invalid operation class '" + opc.type + "': class not found.");
            } catch (OperationException e) {
                throw new RuntimeException(e);
            }
```

Reply:
will let the T&P go through like this first, and will make the change after